src/crimson/os/seastore/async_cleaner.cc

```diff
@@ -514,6 +530,32 @@ journal_seq_t JournalTrimmerImpl::get_tail_limit() const
   return ret;
 }
 
+journal_seq_t JournalTrimmerImpl::get_dirty_tail_min_target() const
+{
+  assert(background_callback->is_ready());
+  auto ret = journal_head.add_offset(
+      backend_type,
+      -static_cast<device_off_t>(config.min_journal_bytes),
+      roll_start,
+      roll_size);
+  return ret;
+}
+
+journal_seq_t JournalTrimmerImpl::get_trim_dirty_target() const
+{
+  assert(background_callback->is_ready());
+  auto journal_dirty_bytes = get_journal_dirty_bytes();
```

Comment:
Can `get_trim_dirty_target()` reuse `get_dirty_bytes_to_trim()` instead?

I think there is another (hidden) invarient in making sure EPM reservation is strictly correct is that, `get_dirty_bytes_to_trim()` called in `do_background_cycle()` is identical to `get_trim_dirty_target()` called in `trim_dirty()`.

Reply:
done:-)